mrobosub_planning/src/captain.py

```diff
@@ -33,13 +35,84 @@
 }
 
 
+def state_class_from_str(full_state: str, transitions: TransitionMap) -> Optional[Type[State]]:
+    """
+    Find the associated class object from a given state name.
+
+    Arguments:
+        full_state (str): The state passed into roslaunch. Could be in the format `state` or `module.state`.
+        transitions (TransitionMap): The transition map from the associated machine name. 
+    
+    Returns:
+        The class object for the state or `None` if no/multiple are found.
+    """
+    if '.' in full_state:
+        # Split the full state into the module.state_name
+        module = full_state.split('.')[0]
+        state = full_state.split('.')[1]
+
+        # We are given a module, search for all of the classes in the module
+        # Note that inspect.getmembers() returns a tuple that looks like (class string name, class object)
+        classes = set(inspect.getmembers(import_module(module), inspect.isclass))
+
+        # Search through all of the tuples by the class string name to 
+        # find associated class object
+        found_class_options = [tup[1] for tup in classes if tup[0] == state]
+
+        # If we find none or multiple, uhh, break
+        if len(found_class_options) != 1:
+            return None
+
+        found_class = found_class_options[0]
+        return found_class
+    else:
+        # Assuming that no module is entered in the terminal, we have to find which module the state is in
+        # set because we want this to be unique
+        module_names = set([cls.__module__ for cls in transitions])
+
+        # We need to get each class for all of the modules included in the transitions module
+        # Note that inspect.getmembers() returns a tuple that looks like (class string name, class object)
+        classes_in_modules = [inspect.getmembers(import_module(mod), inspect.isclass) for mod in module_names]
+
+        # The above list is a list of lists, flatten this into just a list
+        classes_in_transition_map = set([subclass for cls in classes_in_modules for subclass in cls])
+
+        # Search through all of the tuples by the class string name to 
+        # find associated class object
+        found_class_options = [tup[1] for tup in classes_in_transition_map if tup[0] == full_state]
+        
+        # If we find none or multiple, uhh, break
+        if len(found_class_options) != 1:
+            return None
+        
+        found_class = found_class_options[0]
```

Comment:
If the specified state doesn't specify a module (just `StartState` instead of `common_states.StartState`), it would be better if it only searched the states that were in the selected state machine.

Imagine there are two state machines:
```
'aaa': {
    module1.FooState.BarOutcome: Stop,
}
'bbb': {
    module2.FooState.BazOutcome: Stop,
    module1.Bar.Whatever: module2.Bar,
}
```

If I run `roslaunch mrobosub_planning captain.launch machine:=bbb start:=FooState`, I definitely want to run `module2.FooState`, not `module1.FooState`.

If I run `roslaunch mrobosub_planning captain.launch machine:=bbb start:=Bar`, then it should fail since it's ambiguous which `Bar` state I mean.

Reply:
Confusingly, the issue with what you recommend is that there is no easy way to get `StartState` just from the transition map. Just naively looking through all of the classes in the transition map yields you the most specific class e.g. `Complete` for the `default` transition map line `Start.Complete: Submerge`. However because `Complete` is "wrapped" by `StartState`, there is no way of accessing the wrapper class since Python doesn't keep track of this in a `__` method sense (at least to the extent ChatGPT could tell). In this way, it is easy to get the module a class is in (e.g. you can get `common_states` using `cls.__module__`) and the bottom most class (`Complete`) but can't directly access `StartState`. 

That said, I can probably filter all of the `classes_in_transition_map` to ensure that no extra states that are included in that module but not found specifically in the transition map are mapped to extraneously. How does that sound?